### ISSUE TITLE
Cannot have multiple outputs on the same time

### ISSUE THREAD

**Issue by YueXX** [rswestmoreland/graylog-delimited-file-output-plugin#3]
The plug-in cannot have multiple outputs for different streams. If initialize the second one, the first one will stop working

**Comment by MuskratAggie** [comment]
I just ran into this today.  Not only does it stop writing to folder it was, it starts writing to the folder for the stream that the new messages come in to.

**Comment by cukal** [comment]
Yup. Only one output seems possible without upsetting the already configured output, but it's possible to refactor it and build & install it again as a new separate plugin. It really is one the most useful output plugins but I needed several output streams and fixed it this way.

**Comment by mrtndimitrov** [comment]
> Yup. Only one output seems possible without upsetting the already configured output, but it's possible to refactor it and build & install it again as a new separate plugin. It really is one the most useful output plugins but I needed several output streams and fixed it this way.

Hi, how did you refactor it? Changing just the name doesn't seem to work.

**Comment by mrtndimitrov** [comment]
> > Yup. Only one output seems possible without upsetting the already configured output, but it's possible to refactor it and build & install it again as a new separate plugin. It really is one the most useful output plugins but I needed several output streams and fixed it this way.
> 
> Hi, how did you refactor it? Changing just the name doesn't seem to work.

I've changed the code to allow this. It was using static members which were overwritten by the new instances of the plugin. I created a fork (https://github.com/mrtndimitrov/graylog-delimited-file-output-plugin) if someone else wants to allow multiple output files.

**Comment by rswestmoreland** [comment]
> 
> 
> > > Yup. Only one output seems possible without upsetting the already configured output, but it's possible to refactor it and build & install it again as a new separate plugin. It really is one the most useful output plugins but I needed several output streams and fixed it this way.
> > 
> > 
> > Hi, how did you refactor it? Changing just the name doesn't seem to work.
> 
> I've changed the code to allow this. It was using static members which were overwritten by the new instances of the plugin. I created a fork (https://github.com/mrtndimitrov/graylog-delimited-file-output-plugin) if someone else wants to allow multiple output files.

I think this is great, thank you for figuring out how to address this issue.
